### PR TITLE
More custom types

### PR DESCRIPTION
**Pull request by nick-wilson95**
Add custom types for AbilityScripts, DamageData and InputKey

### CODE REVIEW

**Comment by nick-wilson95** [comment]
[Create custom type for oMortal damageList entries](https://trello.com/c/Gfcqbzxo/7-create-custom-type-for-omortal-damagelist-entries)

**Comment by nick-wilson95** [comment]
[Create custom type for oInputManager keyBindings values](https://trello.com/c/9R35mXAk/6-create-custom-type-for-oinputmanager-keybindings-values)

**Comment by nick-wilson95** [comment]
[Create custom types for oAbilityManager abilityToScripts map](https://trello.com/c/YrHTzJA5/8-create-custom-types-for-oabilitymanager-abilitytoscripts-map)

File: scripts/DamageData_GetDamageAmount/DamageData_GetDamageAmount.gml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+/// @function DamageData_getDamageAmount(DamageData_data) Gets the damage amount from an DamageData.
```
Comment (pipkin94) [inline scripts/DamageData_GetDamageAmount/DamageData_GetDamageAmount.gml]: - a DamageData not an

File: scripts/DamageData_GetDamageType/DamageData_GetDamageType.gml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+/// @function DamageData_getDamageType(DamageData_data) Gets the DamageType enum value from an DamageData.
```
Comment (pipkin94) [inline scripts/DamageData_GetDamageType/DamageData_GetDamageType.gml]: a not an

File: scripts/EffectData_getArguments/EffectData_getArguments.gml [hunk 3]
```diff
@@ -3,5 +3,6 @@
 /// @returns {array} The effect arguments.
 var _EffectData_data = argument[0];
 
+if(!EffectData_InstanceOf(_EffectData_data)) ErrorHandler_FatalError("Tried to get arguments from non EffectData structure.");
```
Comment (pipkin94) [inline scripts/EffectData_getArguments/EffectData_getArguments.gml]: Should we write this without indents and { } ?

File: scripts/EffectData_getCasterId/EffectData_getCasterId.gml [hunk 4]
```diff
@@ -3,5 +3,6 @@
 /// @returns {id} The instance id of the caster.
 var _EffectData_data = argument[0];
 
+if(!EffectData_InstanceOf(_EffectData_data)) ErrorHandler_FatalError("Tried to get caster id from non EffectData structure.");
```
Comment (pipkin94) [inline scripts/EffectData_getCasterId/EffectData_getCasterId.gml]: Same again with { }

File: scripts/EffectData_getEffect/EffectData_getEffect.gml [hunk 5]
```diff
@@ -3,5 +3,6 @@
 /// @returns {enum} The effect enum.
 var _EffectData_data = argument[0];
 
+if(!EffectData_InstanceOf(_EffectData_data)) ErrorHandler_FatalError("Tried to get effect from non EffectData structure.");
```
Comment (pipkin94) [inline scripts/EffectData_getEffect/EffectData_getEffect.gml]: { }

File: scripts/EffectData_getSteps/EffectData_getSteps.gml [hunk 6]
```diff
@@ -3,5 +3,6 @@
 /// @returns {number} The effect steps.
 var _EffectData_data = argument[0];
 
+if(!EffectData_InstanceOf(_EffectData_data)) ErrorHandler_FatalError("Tried to get steps from non EffectData structure.");
```
Comment (pipkin94) [inline scripts/EffectData_getSteps/EffectData_getSteps.gml]: { }

**Pull request merged**